src/Raven.Server/Rachis/Follower.cs

```diff
@@ -1127,28 +982,28 @@ private void RequestAllEntries(ClusterOperationContext context, RemoteConnection
                 });
         }
 
-        public void AcceptConnection(LogLengthNegotiation negotiation)
+        public async Task AcceptConnectionAsync(LogLengthNegotiation negotiation)
         {
             if (_engine.CurrentState != RachisState.Passive)
                 _engine.Timeout.Start(_engine.SwitchToCandidateStateOnTimeout);
 
             // if leader / candidate, this remove them from play and revert to follower mode
-            _engine.SetNewState(RachisState.Follower, this, _term,
+            await _engine.SetNewStateAsync(RachisState.Follower, this, _term,
```

Comment:
why async?

Reply:
Cause I call it from async methods so I prefer not to call the sync method (SetNewState).